persistence-api/src/main/java/org/apache/cassandra/stargate/schema/ColumnMetadata.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.cassandra.stargate.schema;
+
+/**
+ * Column information.
+ *
+ * <p>Backend agnostic representation equivalent to {@code
+ * org.apache.cassandra.schema.ColumnMetadata}
+ */
+public interface ColumnMetadata {
+  enum ClusteringOrder {
+    ASC,
+    DESC,
+    NONE
+  }
+
+  enum Kind {
+    PARTITION_KEY,
+    CLUSTERING,
+    REGULAR,
+    STATIC
+  }
+
+  Kind getKind();
+
+  /** Case-sensitive column name. */
+  String getName();
```

Comment:
Maybe we could create a `CqlIdentifier` type - the same that is used in the driver? It is handling quotations, case-sensitivity, etc. I know that adding this dependency may be problematic, but maybe we can port this portion of code to stargate repo?

Reply:
I think that as an end user, it would be easy to reason a case sensitive value: `users_table` or `UserTable`.